src/classes/RD2_OpportunityService_TEST.cls

```diff
@@ -0,0 +1,283 @@
+
+        rd = getRd(rd);
+
+        oppNameFormat = rd.npe03__Donor_Name__c +  ' ' + rd.npe03__Amount__c.format();
+        oppNameFormat += ' ' + CloseDate[0].format() +   ' ' + System.Label.stgNavRecurringDonations;
+        System.assert(rd != null );
+
+        rdsWithCloseDates.put(rd, CloseDate);
+
+        Test.startTest();
+        RD2_OpportunityService rdService = new RD2_OpportunityService();
+        rdService.createInstallmentOpportunities(rdsWithCloseDates);
+        Test.stopTest();
+
+        oppCreated = getOpps();
+        paymentForOpps = getPaymentRecForOpps(oppCreated);
+
+        System.assertEquals(OppCreated[0].npe01__Contact_Id_For_Role__c, primaryContacts[0].id);
+        System.assertEquals(OppCreated[0].Amount, rd.npe03__Amount__c);
+        System.assertEquals(OppCreated[0].Primary_Contact__c, rd.npe03__Contact__c);
+        System.assertEquals(OppCreated[0].Name, oppNameFormat);
+        System.assertEquals(OppCreated[0].closedate, CloseDate[0]);
+        System.assertEquals(OppCreated[0].campaignid, rd.npe03__Recurring_Donation_Campaign__c);
+        System.assertEquals(OppCreated[0].StageName, PLEDGED_STAGENAME);
+        System.assertEquals(paymentForOpps[0].npe01__Payment_Method__c, CREDITCARD);
+
+    }
+
+    /***
+    * @description Verifies if it creates opportunities when an recurring donation
+      is inserted with an account and the stage is Promised.
+    */
+    @isTest
+    private static void shouldCreateOppWhenRDinsertedWithAccountAndAffiliation() {
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Map<npe03__Recurring_Donation__c,List<Date>> rdsWithCloseDates = new Map<npe03__Recurring_Donation__c,List<Date>>();
+        List<Date> CloseDate = new List<Date> ();
+        List<Opportunity> oppCreated = new List<Opportunity> ();
+        List<npe01__OppPayment__c> paymentForOpps = new List<npe01__OppPayment__c> ();
+        String oppNameFormat = '';
+
+
+        CloseDate.add(System.today()+2);
+        CloseDate.add(System.today()+5);
+
+        createRdSettingsWithStage(PROMISED_STAGENAME);
+
+        Account acc = UTIL_UnitTestData_TEST.getOrgAccount();
+        insert acc;
+
+        List<Contact> primaryContacts = UTIL_UnitTestData_TEST.createMultipleTestContacts(1);
+        insert primaryContacts;
+        primaryContacts = getContacts(primaryContacts);
+
+        npe5__Affiliation__c affiliation = new npe5__Affiliation__c(
+            npe5__Organization__c = acc.Id,
+            npe5__Role__c = RELATIONSHIP_TYPE_EMPLOYEE,
+            npe5__Contact__c = primaryContacts[0].Id,
+            npe5__Status__c = RELATIONSHIP_STATUS_CURRENT,
+            Related_Opportunity_Contact_Role__c = AFFILIATION_OCR_ROLE_SOLICITOR
+            );
+
+        insert affiliation;
+
+        Campaign campaign = new Campaign(Name = 'TeaCampaign', IsActive = true);
+        insert campaign;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
```

Comment:
These should use the builder method on the class to construct the right format RD
```suggestion
        npe03__Recurring_Donation__c rd = TEST_RecurringDonationBuilder.constructEnhancedBuilder()
```